src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.CinemaHall;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+public class CinemaHallDaoImpl implements CinemaHallDao {
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
rollback won't work when you are using try with resources because session will be closed before it

Reply:
![image](https://github.com/mate-academy/hibernate-create-movie-session-hw/assets/114025933/ffa65167-4d55-468e-ac1f-2b505445366f)

idea forced me to do it :D

Reply:
Don't be a slave of this technology)))